applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c

```diff
@@ -363,44 +422,56 @@ static int initialize(le_audio_receive_cb recv_cb)
 			/* Channel is not assigned yet: use default */
 			channel = AUDIO_CHANNEL_DEFAULT;
 		}
+
+		for (size_t i = 0; i < ARRAY_SIZE(caps); i++) {
+			ret = bt_audio_capability_register(&caps[i]);
+			if (ret) {
+				LOG_ERR("Capability register failed");
+				return ret;
+			}
+		}
 		if (channel == AUDIO_CH_L) {
 			ret = bt_audio_capability_set_location(BT_AUDIO_DIR_SINK,
 							       BT_AUDIO_LOCATION_FRONT_LEFT);
+			if (ret) {
+				LOG_ERR("Location set failed");
+				return ret;
+			}
+
 		} else {
 			ret = bt_audio_capability_set_location(BT_AUDIO_DIR_SINK,
 							       BT_AUDIO_LOCATION_FRONT_RIGHT);
+			if (ret) {
+				LOG_ERR("Location set failed");
+				return ret;
+			}
 		}
+
+#if CONFIG_STREAM_BIDIRECTIONAL
+		ret = bt_audio_capability_set_location(BT_AUDIO_DIR_SOURCE,
+						       BT_AUDIO_LOCATION_FRONT_LEFT);
 		if (ret) {
 			LOG_ERR("Location set failed");
 			return ret;
 		}
-		ret = bt_audio_capability_register(&caps);
-		if (ret) {
-			LOG_ERR("Capability register failed");
-			return ret;
-		}
-
-		ret = bt_audio_capability_set_available_contexts(
-			BT_AUDIO_DIR_SINK,
-			BT_AUDIO_CONTEXT_TYPE_MEDIA | BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
-		if (ret) {
-			LOG_ERR("Available context set failed");
-			return ret;
+#endif
+		for (size_t i = 0; i < ARRAY_SIZE(streams); i++) {
+			bt_audio_stream_cb_register(&streams[i], &stream_ops);
 		}
 
-		bt_audio_stream_cb_register(&audio_stream, &stream_ops);
 		k_work_init(&adv_work, advertising_process);
 		initialized = true;
 	}
+
 	return 0;
 }
 
 int le_audio_config_get(uint32_t *bitrate, uint32_t *sampling_rate)
 {
-	int frames_per_sec = 1000000 / bt_codec_cfg_get_frame_duration_us(audio_stream.codec);
-	int bits_per_frame = bt_codec_cfg_get_octets_per_frame(audio_stream.codec) * 8;
+	int frames_per_sec = 1000000 / bt_codec_cfg_get_frame_duration_us(streams[0].codec);
```

Comment:
I dont like these hardcoded zeroes here

Reply:
I agree. The way I see it we have two options:

1. Have an array of streams, but have only one sink_stream and one source_stream and then use those for all instances where it is hard-coded
2. Accept that 0 just means the first stream, whether it is source or just any stream

Reply:
Done